app/Http/Livewire/BillTable.php

```diff
@@ -40,12 +41,12 @@ public function query()
     {
         return Bill::query()
             ->when(
-                $this->scope->chosenState(),
-                fn ($query) => $query->where('state_id', $this->scope->chosenState()->id)
+                $this->group->chosenState(),
+                fn ($query) => $query->where('state_id', $this->group->chosenState()->id)
```

Comment:
Don't fix any of this now, just be aware that

- This method does not belong on this class, it should be on `Bill`
- The use of `fn () => ;` throughout severely hurts the readability & clarity of this whole class
- The following makes my OCD go wild:
```
            })
            ->orderByDesc('created')
            ;
```
😱 
- Again, the session use here needs to be ripped out wholesale, so all this can be re-examined as part of that effort.

Just a mental nugget to consider.